core/src/block_creation_loop.rs

```diff
@@ -512,6 +519,20 @@ pub fn start_loop(config: BlockCreationLoopConfig) {
                 break;
             }
 
+            // Check if the entire window is skipped.
+            let highest_parent_ready_slot = leader_window_notifier
```

Comment:
I would move this into `start_leader_retry_replay` so that we can also check this when producing the first block of the leader window.

Also I don't think we need to be this precise, but technically we can give up as soon as we get a skip certificate >= start_slot. Probably more headache than it's worth to additionally track that though.